### DISCUSSION TITLE
How can I register a click/doubleClick on the canvas itself?

### DISCUSSION THREAD

**Discussion by vothvovo** [xyflow/xyflow#3032] [Q&A]
I want to create a node at the place the user doubleClicked the canvas. I tried this:
```
<div
  className="h-full w-full"
  onDoubleClick={() => {
    console.log("double click");
  }}
>
  <BasicFlow />
</div>
```

but it only works if I click on a node, which is the exact opposite of what I want.

**Comment by moklick** [comment]
we don't have onPaneDoubleClick, but you could fake it:

https://codesandbox.io/s/admiring-keldysh-gwfx2q?file=/src/Flow.tsx:1245-1742 (rough sketch)

Does this work for you?
